tests/components/tracked-controls.test.js

```diff
@@ -273,10 +269,129 @@ suite('tracked-controls', function () {
+      assertButtonEvent(emitChangedCalls[0], 'buttonchanged', 0, controller.buttons[0]);
+    });
+
+    test('emits buttonchanged if value changed', function () {
+      const emitSpy = this.sinon.spy(el, 'emit');
+      controller.buttons[0].value = 0.5;
+      component.tick();
+
+      const emitChangedCalls = emitSpy.getCalls().filter(
+        call => call.args[0] === 'buttonchanged');
+      assert.equal(emitChangedCalls.length, 1);
+
+      assertButtonEvent(emitChangedCalls[0], 'buttonchanged', 0, controller.buttons[0]);
+    });
+
+    test('emits independent streams for buttondown, touchstart and buttonchanged', function () {
+      const emitSpy = this.sinon.spy(el, 'emit');
+      controller.buttons[0].pressed = true;
+      controller.buttons[0].touched = true;
+      component.tick();
+
+      // We should emit button, touch and changed calls.
+      assert.equal(emitSpy.getCalls().length, 3);
+      const emitButtonCalls = emitSpy.getCalls().filter(
+        call => call.args[0] === 'buttondown');
+      assert.equal(emitButtonCalls.length, 1);
+
+      const emitTouchCalls = emitSpy.getCalls().filter(
+        call => call.args[0] === 'touchstart');
+      assert.equal(emitTouchCalls.length, 1);
+
+      const emitChangedCalls = emitSpy.getCalls().filter(
+        call => call.args[0] === 'buttonchanged');
+      assert.equal(emitChangedCalls.length, 1);
+
+      assertButtonEvent(emitButtonCalls[0], 'buttondown', 0, controller.buttons[0]);
+      assertButtonEvent(emitTouchCalls[0], 'touchstart', 0, controller.buttons[0]);
+      assertButtonEvent(emitChangedCalls[0], 'buttonchanged', 0, controller.buttons[0]);
+    });
+
+    test('emits independent streams for buttonup, touchend and buttonchanged', function () {
+      const emitSpy = this.sinon.spy(el, 'emit');
+      component.buttonStates[0] = {pressed: true, touched: true, value: 1};
+      controller.buttons[0].pressed = false;
+      controller.buttons[0].touched = false;
+      controller.buttons[0].value = 0;
+      component.tick();
+
+      // Filter down to just our expected events for verification.
+      const emitButtonCalls = emitSpy.getCalls().filter(call => call.args[0] === 'buttonup');
+      assert.equal(emitButtonCalls.length, 1);
+
+      const emitTouchCalls = emitSpy.getCalls().filter(call => call.args[0] === 'touchend');
+      assert.equal(emitTouchCalls.length, 1);
+
+      const emitChangedCalls = emitSpy.getCalls().filter(call => call.args[0] === 'buttonchanged');
+      assert.equal(emitChangedCalls.length, 1);
+
+      // Verify each of the 3 events has the correct event name and state.
+      assertButtonEvent(emitButtonCalls[0], 'buttonup', 0, controller.buttons[0]);
+      assertButtonEvent(emitTouchCalls[0], 'touchend', 0, controller.buttons[0]);
+      assertButtonEvent(emitChangedCalls[0], 'buttonchanged', 0, controller.buttons[0]);
+    });
+
+    test('emits correct event stream for spaced out interaction.', function () {
```

Comment:
Verifies that multi-tick interactions produce the correct series of events for a simple press and release.